frontend/src/components/common/utils.ts

```diff
@@ -1,13 +1,14 @@
-import type { DbViewEnum, SourceLanguage } from "@utils/constants";
+import { DbLanguage } from "@utils/api/types";
+import type { DbViewEnum } from "@utils/constants";
 
 export const getTextPath = ({
-  sourceLanguage,
+  dbLanguage,
   fileName,
   dbView,
 }: {
   fileName?: string;
-  sourceLanguage: SourceLanguage;
+  dbLanguage: DbLanguage | undefined;
```

Comment:
this probably shouldn't allow `undefined`, in this case the url bar will contain the word `undefined`

Reply:
I agree that this is much better than casting! But maybe we should be explicitly throwing an error in this case.

Reply:
I've traced back where this is used and refactored to remove the undefined.

There is of course the same issue with `fileName` which was also originally just cast as a string. I'm drawing my line on this one here, but have left a note so one of us can look at it in the future.